src/preprocessing/passes/unconstrained_simplifier.cpp

```diff
@@ -855,9 +855,15 @@ PreprocessingPassResult UnconstrainedSimplifier::applyInternal(
   {
     processUnconstrained();
     //    d_substitutions.print(Message.getStream());
-    for (Node& assertion : assertions)
+    for (size_t i = 0, asize = assertions.size(); i < asize; ++i)
     {
-      assertion = Rewriter::rewrite(d_substitutions.apply(assertion));
+      Node a = assertions[i];
+      Node as = Rewriter::rewrite(d_substitutions.apply(a));
+      if (a != as)
+      {
+        // replace the assertion
+        assertionsToPreprocess->replace(i, as);
+      }
```

Comment:
This test is already done inside the replace method, so no need to be done outside. The same goes to other occurrences throughout the PR.

Reply:
I'll change, although some are due to debugging traces.